lib/bap/bap.mli

```diff
@@ -608,6 +608,326 @@ module Std : sig
     val pkg_version : string
   end
 
+  module Bytes : sig
+
+    type t = Bytes.t with bin_io, compare, sexp
+
+    include Container.S0   with type t := t with type elt := char
+    include Blit.S         with type t := t
+    include Identifiable.S with type t := t
+    module To_string   : Blit.S_distinct with type src := t with type dst := string
+    module From_string : Blit.S_distinct with type src := string with type dst := t
+
+    external length: t -> int = "%string_length"
+    (** Return the length (number of bytes) of the argument. *)
+
+    external get : t -> int -> char = "%string_safe_get"
+    (** [get s n] returns the byte at index [n] in argument [s].
+
+        Raise [Invalid_argument] if [n] not a valid index in [s]. *)
+
+    external set : t -> int -> char -> unit = "%string_safe_set"
+    (** [set s n c] modifies [s] in place, replacing the byte at index [n]
+        with [c].
+
+        Raise [Invalid_argument] if [n] is not a valid index in [s]. *)
+
+    external create : int -> t = "caml_create_string"
+    (** [create n] returns a new byte sequence of length [n]. The
+        sequence is uninitialized and contains arbitrary bytes.
+
+        Raise [Invalid_argument] if [n < 0] or [n > ]{!Sys.max_string_length}. *)
+
+    val make : int -> char -> t
+    (** [make n c] returns a new byte sequence of length [n], filled with
+        the byte [c].
+
+        Raise [Invalid_argument] if [n < 0] or [n > ]{!Sys.max_string_length}. *)
+
+    val init : int -> (int -> char) -> t
```

Comment:
you should corefy the module, so that it conforms to our coding standards. For example, `init` should have type: `int -> f:(int -> char) -> t`